### ISSUE TITLE
Device hangs randomly

### ISSUE THREAD

**Issue by gismo2004** [j0ta29/esphome#8]
This is basically just for reference for other users.

I had troubles with my ESP stop working/hanging at some point in time. I was not able to reproduce this by switching off/on my Viessmann device, nor was it possible to trigger it by removing the LEDs which was my initial idea of why this happens.

After a while I realized that this problem only occurred, when adding a sensor with an address that can't be read correctly. In my case, it was the error history array (0xA801) .

So in principle it was a configuration issue, but maybe the new VitoWiFi library is more robust against such “stupid users” :-)

**Comment by j0ta29** [comment]
@gismo2004 , can you give me an example of a "stupid user"-configuration (only the part that causes the error). I'll try to reproduce the error.

**Comment by gismo2004** [comment]
I would love to provide this, but now that i have added the message again, for some reason my device is not crashing anymore. So it must have been something else, but i have no clue what it could be... Sorry

**Comment by gismo2004** [comment]
@j0ta29 after some tryouts, I found one configuration that fails. It's just adding a textsensor which does not exist. 

```  - platform: optolink
    id: WPR_GueltBetriebsart_resultHeizKreisanforderung
    name: GültBetriebsart resultierende Heizkreisanforderung
    address: something that does not exist
    bytes: 1
    filters:
      map:
      - 0.0000 -> BA_Standby
      - 1.0000 -> BA_Reduziert
      - 2.0000 -> BA_Normal
      - 3.0000 -> BA_Festwert
 ```
     

this kills my ESP32 randomly

**Comment by j0ta29** [comment]
@gismo2004 , can you still reproduce the problem with the current version of the optlink component?
If yes, can you provide me with the log output before the crash?

**Comment by gismo2004** [comment]
@j0ta29, I can not tell for sure if this is fixed by latest changes or if it is fixed because I have changed my update_intervalls. I had quite some troubles running the latest version, as I got a lot of “queue full” messages when using my previously working config without changing the update_intervall parameters. It worked for a while and then suddenly stops, as I had it before with the "bad" entry. This causes the loop times to vary from 15ms to > 3000ms and this seems to be the main issue. Now with the bad definition from above I also see higher fluctuations in loop times but just up to 60ms which seems to be no problem as it runs now for 2 days without problems. 

Before update:
![image](https://github.com/j0ta29/esphome/assets/2325636/5e5c3491-c9c2-4403-9050-3e94c17a40e1)

while update:
![image](https://github.com/j0ta29/esphome/assets/2325636/403ebddd-d248-4c04-9755-ff029179184d)

After update without "bad" entry:
![image](https://github.com/j0ta29/esphome/assets/2325636/be9e9682-a4bb-43d2-be75-63d12d06c985)

After update with "bad" entry:
![image](https://github.com/j0ta29/esphome/assets/2325636/f6e0e4df-9dad-49c7-a06a-048e14dded24)
